tests/lib/shared/collection-request.js

```diff
@@ -410,6 +410,50 @@ describe( 'CollectionRequest', function() {
 
 		});
 
+		describe( 'year()', function() {
+
+			it( 'should set the "year" filter property on the request object', function() {
+				expect( request.year ).to.exist;
+				expect( request.year ).to.be.a( 'function' );
+				request.year( 2014 );
+				expect( request._filters.year ).to.equal( 2014 );
+			});
+
+		});
+
+		describe( 'month()', function() {
+
+			it( 'should set the "monthnum" filter property on the request object', function() {
+				expect( request.month ).to.exist;
+				expect( request.month ).to.be.a( 'function' );
+				request.month( 7 );
+				expect( request._filters.monthnum ).to.equal( 7 );
+			});
+
+			it( 'should accept month numbers as strings', function() {
+				request.month( '3' );
+				expect( request._filters.monthnum ).to.equal( 3 );
+			});
+
+			it( 'should convert month name strings to month numbers', function() {
+				request.month( 'March' );
+				expect( request._filters.monthnum ).to.equal( 3 );
+				request.month( 'november' );
+				expect( request._filters.monthnum ).to.equal( 11 );
+				request.month( 'Jul' );
+				expect( request._filters.monthnum ).to.equal( 7 );
+			});
+
+		});
+
+		describe( 'day()', function() {
+
+			it( 'should set the "day" filter property on the request object', function() {
+
+			});
```

Comment:
I left this test incomplete, so that you could fill it out -- it should follow the same format as the year test above.